readthedocs/doc_builder/constants.py

```diff
@@ -25,9 +26,43 @@
     )
     DOCKER_IMAGE_SETTINGS.update(old_config)
 
-DOCKER_LIMITS = {'memory': '200m', 'time': 600}
+DOCKER_LIMITS = {
+    'memory': '200m',
+    'time': 600,
+}
+
+# Set docker limits dynamically based on system memory
+# This assumes 1-builder per server
+try:
+    total_memory = int(subprocess.check_output("free -m | awk '/^Mem:/{print $2}'", shell=True))
```

Comment:
We could use psutil here: https://pypi.org/project/psutil/ -- but it would add a dependency. Not sure the tradeoff.

Reply:
I'd worry a little about this call failing for some reason, then falling through to the except because it's a generic Exception, and then we could be in a place where we are setting the default amount on servers without noticing.

`psutil` might give us more confidence in this check, or the except should be much tighter. The failure case is likely MacOS doing something different, or MacOS free/awk doing something different than Linux free/gnu awk, so we could also put this default set operation in a `if DEBUG` inside a `@property` setting. We likely never want to settle on the defaults in prod